### ISSUE TITLE
Listing apps fails, due to the recent pricing update on App Store?

### ISSUE THREAD

**Issue by akkyie** [fastlane/fastlane#21125]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

When I run the action `app_store_build_number`, it fails with an error response from Connect API: `The request cannot be fulfilled because of the state of another resource. - The resource 'appPrices' cannot be viewed, created or updated. Please view and create 'manualPrices' using the resource 'appPriceSchedules'.`

<details>
<summary><code>fastlane summary</code> and stack trace</summary>

```
+------+----------------------------------+-------------+
|                   fastlane summary                    |
+------+----------------------------------+-------------+
| Step | Action                           | Time (in s) |
+------+----------------------------------+-------------+
| 1    | Switch to get_apple_api_key lane | 0           |
| 2    | app_store_connect_api_key        | 0           |
| 💥   | app_store_build_number           | 0           |
+------+----------------------------------+-------------+

[14:47:58]: fastlane finished with errors
bundler: failed to load command: fastlane (/Users/distiller/project/vendor/bundle/ruby/3.0.0/bin/fastlane)
/Users/distiller/.rbenv/versions/3.0.5/bin/bundle: \e[31m[!] The request cannot be fulfilled because of the state of another resource. - The resource 'appPrices' cannot be viewed, created or updated. Please view and create 'manualPrices' using the resource 'appPriceSchedules'.\e[0m (Spaceship::UnexpectedResponse)
/Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/api_client.rb:204:in `handle_response': The request cannot be fulfilled because of the state of another resource. - The resource 'appPrices' cannot be viewed, created or updated. Please view and create 'manualPrices' using the resource 'appPriceSchedules'. (Spaceship::UnexpectedResponse)
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/api_client.rb:113:in `get'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/testflight/testflight.rb:22:in `get_apps'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/3.0.0/forwardable.rb:238:in `get_apps'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/models/app.rb:75:in `all'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/models/app.rb:81:in `find'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/app_store_build_number.rb:53:in `get_build_info'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/app_store_build_number.rb:23:in `get_build_version_and_number'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/app_store_build_number.rb:12:in `run'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:263:in `block (2 levels) in execute_action'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/actions_helper.rb:69:in `execute_action'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:255:in `block in execute_action'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:229:in `chdir'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:229:in `execute_action'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:157:in `trigger_action_by_name'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/fast_file.rb:159:in `method_missing'
	from Fastfiles/ReleaseFastfile.rb:96:in `block in parsing_binding'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/commands_generator.rb:110:in `block (2 levels) in run'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/commands_generator.rb:354:in `run'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/commands_generator.rb:43:in `start'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/cli_tools_distributor.rb:123:in `take_off'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/gems/fastlane-2.212.1/bin/fastlane:23:in `<top (required)>'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/bin/fastlane:25:in `load'
	from /Users/distiller/project/vendor/bundle/ruby/3.0.0/bin/fastlane:25:in `<top (required)>'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:58:in `load'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:58:in `kernel_load'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli/exec.rb:23:in `run'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli.rb:483:in `exec'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli.rb:31:in `dispatch'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/cli.rb:25:in `start'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/exe/bundle:48:in `block in <top (required)>'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
	from /Users/distiller/.rbenv/versions/3.0.5/lib/ruby/gems/3.0.0/gems/bundler-2.3.13/exe/bundle:36:in `<top (required)>'
	from /Users/distiller/.rbenv/versions/3.0.5/bin/bundle:23:in `load'
	from /Users/distiller/.rbenv/versions/3.0.5/bin/bundle:23:in `<main>'

Exited with code exit status 1
```

</details>

As far as I look over the stack trace, I suppose it's caused by the recent pricing change on App Store and the related Connect API update.
https://developer.apple.com/news/releases/?id=03092023b

The `prices` attribute included in `ESSENTIAL_INCLUDES`:
https://github.com/fastlane/fastlane/blob/e55138d9dc769654897cf95701accc698d8edb01/spaceship/lib/spaceship/connect_api/models/app.rb#L60-L63

... looks now marked as deprecated.
https://developer.apple.com/documentation/appstoreconnectapi/app/relationships

### Environment

```
gem "fastlane", "2.212.1"
```

<details><summary>✅ fastlane environment ✅</summary>

This is my local environment and it might be not so meaningful as the error above happened on my CI environment

### Stack

| Key                         | Value                                                                       |
| --------------------------- | --------------------------------------------------------------------------- |
| OS                          | 13.1                                                                        |
| Ruby                        | 3.0.3                                                                       |
| Bundler?                    | true                                                                        |
| Git                         | git version 2.39.2                                                          |
| Installation Source         | *hidden* |
| Host                        | macOS 13.1 (22C65)                                                          |
| Ruby Lib Dir                | ~/.rbenv/versions/3.0.3/lib                                                 |
| OpenSSL Version             | OpenSSL 1.1.1q  5 Jul 2022                                                  |
| Is contained                | false                                                                       |
| Is homebrew                 | false                                                                       |
| Is installed via Fabric.app | false                                                                       |
| Xcode Path                  | /Applications/Xcode-14.2.0.app/Contents/Developer/                          |
| Xcode Version               | 14.2                                                                        |
| Swift Version               | 5.7.2                                                                       |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
import "Fastfiles/HelperFastfile.rb"
import "Fastfiles/ReleaseFastfile.rb"
import "Fastfiles/SandboxFastfile.rb"
import "Fastfiles/AppDistributionFastfile.rb"
import "Fastfiles/LicenseFastfile.rb"

before_all do
  ENV['TZ'] = 'Asia/Tokyo'
  ENV["REPO"] = "abema/abema-ios"
  ENV["SLACK_URL"] = ENV["SLACK_GENERAL_WEBHOOK"]
  ENV["SPACESHIP_ONLY_ALLOW_INTERACTIVE_2FA"] = "true"
  ENV["FASTLANE_IS_INTERACTIVE"] = "false"
  ENV["CI"]= "1"
end

```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.212.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

| Plugin                                    | Version | Update-Status |
| ----------------------------------------- | ------- | ------------- |
| fastlane-plugin-firebase_app_distribution | 0.5.0   | ✅ Up-To-Date  |

<details><summary><b>Loaded gems</b></summary>

| Gem                                       | Version      |
| ----------------------------------------- | ------------ |
| did_you_mean                              | 1.5.0        |
| bundler                                   | 2.3.13       |
| pathname                                  | 0.1.0        |
| rake                                      | 13.0.6       |
| rexml                                     | 3.2.5        |
| CFPropertyList                            | 3.0.6        |
| concurrent-ruby                           | 1.2.0        |
| i18n                                      | 1.12.0       |
| minitest                                  | 5.17.0       |
| tzinfo                                    | 2.0.6        |
| zeitwerk                                  | 2.6.6        |
| activesupport                             | 6.1.7.2      |
| public_suffix                             | 4.0.7        |
| addressable                               | 2.8.1        |
| httpclient                                | 2.8.3        |
| json                                      | 2.6.3        |
| algoliasearch                             | 1.27.5       |
| artifactory                               | 3.0.15       |
| atomos                                    | 0.1.3        |
| aws-eventstream                           | 1.2.0        |
| aws-partitions                            | 1.716.0      |
| aws-sigv4                                 | 1.5.2        |
| jmespath                                  | 1.6.2        |
| aws-sdk-core                              | 3.170.0      |
| aws-sdk-kms                               | 1.62.0       |
| aws-sdk-s3                                | 1.119.1      |
| babosa                                    | 1.0.4        |
| claide                                    | 1.1.0        |
| colored2                                  | 3.1.2        |
| cork                                      | 0.3.0        |
| nap                                       | 1.1.0        |
| open4                                     | 1.3.4        |
| claide-plugins                            | 0.9.2        |
| fuzzy_match                               | 2.0.4        |
| netrc                                     | 0.11.0       |
| ffi                                       | 1.15.5       |
| ethon                                     | 0.15.0       |
| typhoeus                                  | 1.4.0        |
| cocoapods-core                            | 1.11.3       |
| cocoapods-deintegrate                     | 1.0.5        |
| cocoapods-downloader                      | 1.6.3        |
| cocoapods-plugins                         | 1.0.0        |
| cocoapods-search                          | 1.0.1        |
| cocoapods-trunk                           | 1.6.0        |
| cocoapods-try                             | 1.2.0        |
| escape                                    | 0.0.4        |
| fourflusher                               | 2.3.1        |
| gh_inspector                              | 1.1.3        |
| molinillo                                 | 0.8.0        |
| ruby-macho                                | 2.5.1        |
| nanaimo                                   | 0.3.0        |
| xcodeproj                                 | 1.22.0       |
| cocoapods                                 | 1.11.3       |
| colored                                   | 1.2          |
| highline                                  | 2.0.3        |
| commander                                 | 4.6.0        |
| faraday-em_http                           | 1.0.0        |
| faraday-em_synchrony                      | 1.0.0        |
| faraday-excon                             | 1.1.0        |
| faraday-httpclient                        | 1.0.1        |
| multipart-post                            | 2.0.0        |
| faraday-multipart                         | 1.0.4        |
| faraday-net_http                          | 1.0.1        |
| faraday-net_http_persistent               | 1.2.0        |
| faraday-patron                            | 1.0.0        |
| faraday-rack                              | 1.0.0        |
| faraday-retry                             | 1.0.3        |
| ruby2_keywords                            | 0.0.5        |
| faraday                                   | 1.10.3       |
| faraday-http-cache                        | 2.4.1        |
| rchardet                                  | 1.8.0        |
| git                                       | 1.13.1       |
| kramdown                                  | 2.4.0        |
| kramdown-parser-gfm                       | 1.1.0        |
| no_proxy_fix                              | 0.1.2        |
| sawyer                                    | 0.9.2        |
| octokit                                   | 5.6.1        |
| unicode-display_width                     | 1.8.0        |
| terminal-table                            | 1.8.0        |
| danger                                    | 9.2.0        |
| danger-plugin-api                         | 1.0.0        |
| danger-swiftformat                        | 0.8.1        |
| xcresult                                  | 0.2.1        |
| danger-xcode_summary                      | 1.2.0        |
| declarative                               | 0.0.20       |
| digest-crc                                | 0.6.4        |
| unf_ext                                   | 0.0.8.2      |
| unf                                       | 0.1.4        |
| domain_name                               | 0.5.20190701 |
| dotenv                                    | 2.8.1        |
| emoji_regex                               | 3.2.3        |
| excon                                     | 0.99.0       |
| http-cookie                               | 1.0.5        |
| faraday-cookie_jar                        | 0.0.7        |
| faraday_middleware                        | 1.2.0        |
| fastimage                                 | 2.2.6        |
| jwt                                       | 2.7.0        |
| memoist                                   | 0.16.2       |
| multi_json                                | 1.15.0       |
| os                                        | 1.1.4        |
| signet                                    | 0.17.0       |
| googleauth                                | 1.3.0        |
| mini_mime                                 | 1.1.2        |
| trailblazer-option                        | 0.1.2        |
| uber                                      | 0.1.0        |
| representable                             | 3.2.0        |
| retriable                                 | 3.1.2        |
| webrick                                   | 1.8.1        |
| google-apis-core                          | 0.11.0       |
| google-apis-androidpublisher_v3           | 0.34.0       |
| google-apis-playcustomapp_v1              | 0.13.0       |
| google-apis-iamcredentials_v1             | 0.17.0       |
| google-apis-storage_v1                    | 0.19.0       |
| google-cloud-env                          | 1.6.0        |
| google-cloud-errors                       | 1.3.0        |
| google-cloud-core                         | 1.6.0        |
| google-cloud-storage                      | 1.44.0       |
| mini_magick                               | 4.12.0       |
| naturally                                 | 2.2.1        |
| optparse                                  | 0.1.1        |
| plist                                     | 3.7.0        |
| rubyzip                                   | 2.3.2        |
| security                                  | 0.1.3        |
| simctl                                    | 1.6.10       |
| terminal-notifier                         | 2.0.0        |
| tty-screen                                | 0.8.1        |
| tty-cursor                                | 0.7.1        |
| tty-spinner                               | 0.9.3        |
| word_wrap                                 | 1.0.0        |
| rouge                                     | 2.0.7        |
| xcpretty                                  | 0.3.0        |
| xcpretty-travis-formatter                 | 1.0.1        |
| fastlane-plugin-firebase_app_distribution | 0.5.0        |
</details>

*generated on:* **2023-03-10**
</details>

**Comment by Glaubenio** [comment]
Same issue here

**Comment by tymcdowell** [comment]
Same here.  We cannot upload using Fastlane.

**Comment by joshstrange** [comment]
Just ran into this issue as well

**Comment by joshstrange** [comment]
Ok I'm able to run `pilot` successfully after commenting out the "prices" array element. I have the following patchfile (`fastlane-prices-fix.patch`) in the root folder of my fastlane-related code (where I install my gems):

```patch
--- vendor/bundle/ruby/3.2.0/gems/fastlane-2.211.0/spaceship/lib/spaceship/connect_api/models/app-original.rb	2023-03-10 18:42:18.000000000 -0500
+++ vendor/bundle/ruby/3.2.0/gems/fastlane-2.211.0/spaceship/lib/spaceship/connect_api/models/app.rb	2023-03-10 18:42:39.000000000 -0500
@@ -59,7 +59,7 @@
 
       ESSENTIAL_INCLUDES = [
         "appStoreVersions",
-        "prices"
+#        "prices"
       ].join(",")
 
       def self.type

```

And I added the following to my GH Actions workflow:

```
      - name: Patch Fastlane
        run: |
          patch vendor/bundle/ruby/3.2.0/gems/fastlane-2.211.0/spaceship/lib/spaceship/connect_api/models/app.rb fastlane-prices-fix.patch
```

Obviously this only works if you use ruby 3.2.0 and fastlane 2.11.0 but the basic idea should work for any version of either.

I created the patch by starting in root of my project (after I copied `app.rb` to `app-original.rb` and commented out the `"prices"` line in `app.rb`) and running:

```bash
diff -Naur vendor/bundle/ruby/3.2.0/gems/fastlane-2.211.0/spaceship/lib/spaceship/connect_api/models/app-original.rb vendor/bundle/ruby/3.2.0/gems/fastlane-2.211.0/spaceship/lib/spaceship/connect_api/models/app.rb > fastlane-prices-fix.patch
```

(Again, adjust for ruby/fastlane version).

Nothing I did here is setting the world on fire but it unblocked me and so I wanted to share in case it helps someone else who was planning on using fastlane until this is fixed.

**Comment by jffmrk** [comment]
Thanks @joshstrange - that worked for me too. I forked the repo, committed the change, and updated `Gemfile` to use my forked repo.

**Comment by manucodin** [comment]
Same here, im using fastlane with my CI environment and i cannot upload versions

**Comment by wow-such-amazing** [comment]
We are getting the same error today ➕

**Comment by CheeryLee** [comment]
Same error on 2.212.1 too.
@joshstrange's workaround helps out.

**Comment by ploth** [comment]
Same here with ruby 3.0.0 and fastlane 2.212.1. @joshstrange's workaround helps out. 😘

**Comment by sck-v** [comment]
Just use this in `Fastfile`. It will give you warning, that the constant is redefined, but you don't need to maintain diffs and patches
```ruby
Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
```

**Comment by frenkyoptic** [comment]
@sck-v could you provide more details please? i put this into Fastfile, but no changes, still issue

**Comment by sck-v** [comment]
>  but no changes, still issue

Maybe you would also need to `require 'spaceship'` before that line. We have quite complex fastlane setup, so this may be the key as well

**Comment by maxTeste** [comment]
@sck-v any idea how can I try that solution on azure devops? 
![image](https://user-images.githubusercontent.com/75133022/225882597-bb30a9b8-ef40-48d9-ac62-94f375c354fe.png)


I create all these steps in the pipeline


![image](https://user-images.githubusercontent.com/75133022/225882702-88ec775d-c229-476a-97bb-852d35485159.png)

**Comment by xinatcg** [comment]
> Just use this in `Fastfile`. It will give you warning, that the constant is redefined, but you don't need to maintain diffs and patches
> 
> ```ruby
> Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> ```

base on our test this solution works,

```
require 'spaceship'
Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
```

add these 2 lines in the top of Fastline file

```
# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane
require 'spaceship'
Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
// THE OTHER CONTENT

```

**Comment by frenkyoptic** [comment]
For me, this was working only from Appfile

**Comment by tbenyon** [comment]
> > Just use this in `Fastfile`. It will give you warning, that the constant is redefined, but you don't need to maintain diffs and patches
> > ```ruby
> > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > ```
> 
> base on our test this solution works,
> 
> ```ruby
> require 'spaceship'
> Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> ```
> 
> add these 2 lines in the top of Fastline file
> 
> ```
> # Uncomment the line if you want fastlane to automatically update itself
> # update_fastlane
> require 'spaceship'
> Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> // THE OTHER CONTENT
> ```

We tried this but are now getting this error:
```
bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
/Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.2/deliver/lib/deliver/upload_price_tier.rb:22:in `upload': [31m[!] undefined method `first' for nil:NilClass[0m (NoMethodError)
```

**Comment by xinatcg** [comment]
> > > Just use this in `Fastfile`. It will give you warning, that the constant is redefined, but you don't need to maintain diffs and patches
> > > ```ruby
> > > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > > ```
> > 
> > 
> > base on our test this solution works,
> > ```ruby
> > require 'spaceship'
> > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > add these 2 lines in the top of Fastline file
> > ```
> > # Uncomment the line if you want fastlane to automatically update itself
> > # update_fastlane
> > require 'spaceship'
> > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > // THE OTHER CONTENT
> > ```
> 
> We tried this but are now getting this error:
> 
> ```
> bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
> /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.2/deliver/lib/deliver/upload_price_tier.rb:22:in `upload': [31m[!] undefined method `first' for nil:NilClass[0m (NoMethodError)
> ```

not sure, maybe the version?  we are on 2.211.0, maybe you can try upgrade
```
# fastlane 2.212.1 is available. You are on 2.211.0.
```

**Comment by jhong70** [comment]
> > > Just use this in `Fastfile`. It will give you warning, that the constant is redefined, but you don't need to maintain diffs and patches
> > > ```ruby
> > > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > > ```
> > 
> > 
> > base on our test this solution works,
> > ```ruby
> > require 'spaceship'
> > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > add these 2 lines in the top of Fastline file
> > ```
> > # Uncomment the line if you want fastlane to automatically update itself
> > # update_fastlane
> > require 'spaceship'
> > Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> > // THE OTHER CONTENT
> > ```
> 
> We tried this but are now getting this error:
> 
> ```
> bundler: failed to load command: fastlane (/usr/local/bin/fastlane)
> /Library/Ruby/Gems/2.6.0/gems/fastlane-2.205.2/deliver/lib/deliver/upload_price_tier.rb:22:in `upload': [31m[!] undefined method `first' for nil:NilClass[0m (NoMethodError)
> ```

https://github.com/fastlane/fastlane/blob/e55138d9dc769654897cf95701accc698d8edb01/deliver/lib/deliver/upload_price_tier.rb#L19-L32

I have a similar issue for a completely new app we're building where prices haven't been set before. I edited the code to return early and seems to have fixed the issue. The correct solution is to update fastlane to handle the new price App Store Connect API updates.

`
if app_prices.nil?
  UI.message("App has no prices yet... No changes required."
return
end
`

**Comment by sck-v** [comment]
> I have a similar issue for a completely new app we're building where prices haven't been set before. I edited the code to return early and seems to have fixed the issue. The correct solution is to update fastlane to handle the new price App Store Connect API updates.

I bet monkeypatch may help here as well. Something like this should fix the exception

```ruby
Spaceship::ConnectAPI::App.class_eval do 
  def prices 
    []
  end
end
```

**Comment by Jarutais** [comment]
I'm having this problem when trying to setup Fastlane to a new app. Fastlane init fails with this error.

**Comment by zachawilson** [comment]
I also had the nil app_prices issue and included that patch in my fork.

The monkey patching didn't work for me. I am not sure why, but I will comment that we are pulling in our own fastlane plugin to iterate on our custom logic and share across our organization. When I attempted the money patching, I did it in the main Fastfile and not in the plugin.

**Comment by tbenyon** [comment]
Nothing worked for us but we were only seeing the issue on new apps.

We manually modified the gem package code is the system files for now (as per [this](https://github.com/Jarutais/fastlane/commit/236713ec50046fb8cb771cd0cd666ec039832df5)) as we only have one build machine. (NASTY WE KNOW!)

Hope to see a fix in the next Fastlane update and then will go from there

**Comment by SamuelGadiel** [comment]
> ```diff
>   ESSENTIAL_INCLUDES = [
>          "appStoreVersions",
> -        "prices"
> +#        "prices"
>        ].join(",")
>  
>        def self.type
> ```

This is awesome, worked as a charm!
But what I found interesting is that it happened two cases with me

The first one I've just created an app in the app store
And without any further configuration made the upload with fastlane
It worked normally, as usual
Notice that in this case I hadn't configured the price yet

In the second app, I've created the app in the app store
But this time I started to configure the store with necessary information, including the prices
Only then I proceed to submit the app with fastlane, and in this case it showed the above error

So it seems that if you set the prices before submiting the app this error occurs, but if you skip the prices fastlane can upload the app without any issues. After that you can set the prices with no problem

**Comment by joshstrange** [comment]
> The first one I've just created an app in the app store
> And without any further configuration made the upload with fastlane
> It worked normally, as usual
> Notice that in this case I hadn't configured the price yet

Yep, I saw the exact same behavior. My unreleased app (with no price configured) had no issues while my other apps did. Once I set a price it started having the issue.

**Comment by nonib** [comment]
Hi, 
I was facing the same issue while i was trying to upload apps threw "Azure Devops", some succeeded and some failed with the error above.
i have succeeded to overcome this by doing the following:
* first upload the package to test-flight using the altool (manually or by cicd):
xcrun altool --upload-app -f "{ipaPath}" -t ios --apiKey {Key} --apiIssuer {IssuerId}
* then you will see the bundle assigned to the test-flight
* next uploads can be done using the regular cicd job "AppStoreRelease@1" with your properties.

hope it will help somebody...

**Comment by gelodgreat** [comment]
> ```ruby
> require 'spaceship'
> Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')
> ```

this fixes the issue for me, though I believe Fastlane team should also put an update fix for this as this is breaking the flow.

**Comment by Stypox** [comment]
In case you are using `bundle`, this command will apply the patch without having to specify Ruby and/or Fastlane versions:
```bash
echo -e "import sys\nv=open(sys.argv[1]).read()\nv=v.replace(',\\\n        \"prices\"\\\n', '\\\n')\nwith open(sys.argv[1], 'w') as f:\n    f.write(v)" | python3 - $(bundle show fastlane)/spaceship/lib/spaceship/connect_api/models/app.rb
```

The command above runs both on Linux and on MacOS. In case you need a Linux-only command that does not use `python3`, here it is:
```bash
cat <<< $(cat $(bundle show fastlane)/spaceship/lib/spaceship/connect_api/models/app.rb | tr '\n' '\r' | sed -E 's/,\r        "prices"\r/\r/' | tr '\r' '\n') > $(bundle show fastlane)/spaceship/lib/spaceship/connect_api/models/app.rb
```

**Comment by bondarenkod** [comment]
I am pleased to report that the ipa file has been successfully uploaded to the app store (TestFlight) via iTMSTransporter, utilizing the 'new' Transporter app. The following PowerShell script was employed:

```
$iTMSTransporter = "/Applications/Transporter.app/Contents/itms/bin/iTMSTransporter"
$login = "yor-apple-id@login.com";
$pwd = "application-specific-pwd";
$pkg = "./path.to.the.ipa";
$bundle = "path.to.the";
$provider = "my team id";

# & $iTMSTransporter -m provider -u $login -p $pwd <- use this command to get your team id

& $iTMSTransporter -m upload -u $login -p $pwd -asc_provider $provider -assetFile $pkg
```

**Comment by MomenYasser** [comment]
Any update here guys? I'm facing same problem on Azure DevOps

<img width="1495" alt="image" src="https://user-images.githubusercontent.com/6758791/229276553-783c2121-4922-422d-9425-a73e531b04d8.png">

<img width="1492" alt="image" src="https://user-images.githubusercontent.com/6758791/229276564-29227c79-2e2d-4a84-9e5f-e204effb59d4.png">

**Comment by MomenYasser** [comment]
Guys I confirm it's worked fine with me once I added app ID (You can find it in app information from Apple Store Connect)
<img width="213" alt="image" src="https://user-images.githubusercontent.com/6758791/229286312-c1031d67-3a01-4c96-9383-34d6825a7e83.png">

**Comment by ff-ahumberston** [comment]
Applying the patch from @joshstrange, problem resolved.

Does anybody know if this has been pushed to the main release?

**Comment by bondarenkod** [comment]
@MomenYasser check my comment, it's working.

**Comment by evgenyvalavin** [comment]
Same for me. Azure DevOps

**Comment by lucasGabrielDeAA** [comment]
I am facing the same error. Any news?
Using a patched version nor a forked fastlane version is not practicable in my case

**Comment by Michelalmeidasilva** [comment]
Same problem for me.

**Comment by carloslibardo** [comment]
Same problem here, using bitrise :c

**Comment by Gambitboy** [comment]
Any workaround for bitrise or fix being deployed?

We're using https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver step. And it started failing about a week or 2 ago.

**Comment by deiv061** [comment]
For Azure DevOps, with hosted macOS agent seems to fix that with  shell exec step, after installation on fastlane
```
sudo sed -i ' ' '61d' /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/models/app.rb
sudo sed -i ' ' '61s/.*/"appStoreVersions"/' /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.212.1/spaceship/lib/spaceship/connect_api/models/app.rb
```

**Comment by joshdholtz** [comment]
Ahhh, not sure how I missed this issue 😱  Will try to look into later today! So so sorry, everyone 😔

**Comment by h-tko** [comment]
In my case, I was able to temporarily solve it by commenting out using sed as follows.

```
find ./ -name app.rb | grep connect_api | xargs sed -i -e 's/\"prices/#\"prices/g'
```

**Comment by giginet** [comment]
I fixed this issue on https://github.com/fastlane/fastlane/pull/21187.

It will be released on the next release 🚀

**Comment by MiguelSOliveira** [comment]
Tried to apply the changes from the PR above but getting this error:

```
bundler: failed to load command: fastlane (/app/ios/vendor/bundle/ruby/2.6.0/bin/fastlane)
NoMethodError: [!] undefined method `first' for nil:NilClass
  /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/upload_price_tier.rb:24:in `upload'
  /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/runner.rb:161:in `upload_metadata'
  /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/runner.rb:61:in `run'
  /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/upload_to_app_store.rb:22:in `run'
```

Not sure if I'm the only one?

**Comment by 0011001011** [comment]
> 

Same problem here

**Comment by 0011001011** [comment]
> Tried to apply the changes from the PR above but getting this error:
> 
> ```
> bundler: failed to load command: fastlane (/app/ios/vendor/bundle/ruby/2.6.0/bin/fastlane)
> NoMethodError: [!] undefined method `first' for nil:NilClass
>   /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/upload_price_tier.rb:24:in `upload'
>   /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/runner.rb:161:in `upload_metadata'
>   /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/deliver/lib/deliver/runner.rb:61:in `run'
>   /app/ios/vendor/bundle/ruby/2.6.0/gems/fastlane-2.212.1/fastlane/lib/fastlane/actions/upload_to_app_store.rb:22:in `run'
> ```
> 
> Not sure if I'm the only one?

Same here

**Comment by 0011001011** [comment]
> > Not sure if I'm the only one?

I tried multiple monkey patching but always end up with some error

Latest try, based on @giginet https://github.com/giginet/fastlane/pull/1/commits

Results in an API error : 
`The request cannot be fulfilled because of the state of another resource. - The resource 'appPrices' cannot be viewed, created or updated. Please view and create 'manualPrices' using the resource 'appPriceSchedules'`

**Comment by MiguelSOliveira** [comment]
My app was already created but I did do a manual change when apple brought these changes. So I guess it's not so much "just created" but "updated manually" that could be causing the problem

**Comment by nadicadimitrijevska-allocate** [comment]
@MiguelSOliveira - I have the same issue. The app where I did a manual change has the same issue.

**Comment by via-guy** [comment]
We use @joshstrange's script above to patch Fastlane, which we install through a `Gemfile` and `bundle install --deployment`, to find the right file and patch it. Maybe someone can improve on it, but it works great for us.

```sh
#!/bin/sh

FASTLANE_DIR=`find vendor -type d -regex ".*/fastlane-[0-9]*\.[0-9]*\.[0-9]*"`

echo "Patching fastlane in $FASTLANE_DIR"

# https://github.com/fastlane/fastlane/issues/21125#issuecomment-1464684522
cat << EOF > .fastlane-prices-fix.patch
--- $FASTLANE_DIR/spaceship/lib/spaceship/connect_api/models/app-original.rb   2023-03-10 18:42:18.000000000 -0500
+++ $FASTLANE_DIR/spaceship/lib/spaceship/connect_api/models/app.rb    2023-03-10 18:42:39.000000000 -0500
@@ -59,7 +59,7 @@

       ESSENTIAL_INCLUDES = [
         "appStoreVersions",
-        "prices"
+#        "prices"
       ].join(",")

       def self.type
EOF

patch -N $FASTLANE_DIR/spaceship/lib/spaceship/connect_api/models/app.rb .fastlane-prices-fix.patch
rm .fastlane-prices-fix.patch
```

**Comment by lucasGabrielDeAA** [comment]
> I fixed this issue on #21187.
> 
> It will be released on the next release 🚀

Hi guys!
Was this issue already reviewed and merged? I've been seeing a lot of people still facing the same problem.
Specially those who can't use a patched version of fastlane due to some CI's behavior

**Comment by lucas-koontz** [comment]
Not sure if anyone pointed it out, but [this method ](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_price_tier.rb#L7 )also has a problem caused by this same issue (deprecation of App <-> Prices relationship on App Store Connect API).

Specifically, these [two lines ](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_price_tier.rb#L23-L24)will produce a nil pointer error because apps do not have the price property anymore.

**Comment by MarkCSpike** [comment]
> I fixed this issue on #21187.
> 
> It will be released on the next release 🚀

Any timescale on when this fix will be released? Unfortunately we can't patch our CI/CD and it is breaking some of our builds.

**Comment by SteveKennedy** [comment]
This item seems like a complete show stopper for any CI/CD process using Fastlane to deploy to Apple.

None of the workarounds suggested up to this point work any more.

**Comment by mjvestal** [comment]
> Not sure if anyone pointed it out, but [this method ](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_price_tier.rb#L7)also has a problem caused by this same issue (deprecation of App <-> Prices relationship on App Store Connect API).
> 
> Specifically, these [two lines ](https://github.com/fastlane/fastlane/blob/master/deliver/lib/deliver/upload_price_tier.rb#L23-L24)will produce a nil pointer error because apps do not have the price property anymore.

To @lucas-koontz 's point, creating a new app with `deliver` `price_tier: 0` started throwing a 500 error:
```bash


[13:36:21]: Successfully uploaded screenshots to App Store Connect
--
  | [13:36:21]: App has no prices yet... Enabling all countries in App Store Connect



+------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+
--
  | \|                                                                                        Lane Context                                                                                        \|
  | +------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+

....



+------------------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------+
--
  | [13:36:22]: Server error got 500
```

I manually set pricing and availability in ASC and reran the build but now it fails with the error in the original post
```bash
[14:36:11]: The request cannot be fulfilled because of the state of another resource. - The resource 'appPrices' cannot be viewed, created or updated. Please view and create 'manualPrices' using the resource 'appPriceSchedules'.
```

**Comment by Abdelazeem777** [comment]
Any updates?

**Comment by MiguelSOliveira** [comment]
Is this closed because of the merged PR? Will the other issue be handled in a new issue? How can we track it? @joshdholtz

**Comment by gsantner** [comment]
Just if somebody is about to try the latest release: As the bug is closed and version 2.212.2 was released yesterday, thought it is fixed. Unfortunately deployment still fails with the `appPrices` error.

**Comment by tat200809** [comment]
> Just if somebody is about to try the latest release: As the bug is closed and version 2.212.2 was released yesterday, thought it is fixed. Unfortunately deployment still fails with the `appPrices` error.

Tested as well, can confirm that the issue still persists

**Comment by Gambitboy** [comment]
I'm using a bitrise step that uses fastlane and I'm still getting a crash.

Not sure if its a bug in fastlane or in the bitrise step

```
[10:27:37]: wrong number of arguments (given 2, expected 1)
/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/rubyzip-3.0.0.alpha/lib/zip/file.rb:92:in `open'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:68:in `fetch_info_plist_with_rubyzip'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:41:in `fetch_info_plist_file'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane_core/lib/fastlane_core/ipa_file_analyser.rb:18:in `fetch_app_version'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/detect_values.rb:73:in `find_version'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/detect_values.rb:20:in `run!'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/runner.rb:25:in `initialize'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/commands_generator.rb:82:in `new'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/commands_generator.rb:82:in `block (2 levels) in run'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:187:in `call'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/command.rb:157:in `run'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/runner.rb:444:in `run_active_command'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:124:in `run!'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/commander-4.6.0/lib/commander/delegates.rb:18:in `run!'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/commands_generator.rb:187:in `run'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/deliver/lib/deliver/commands_generator.rb:18:in `start'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/cli_tools_distributor.rb:115:in `take_off'\n/Users/vagrant/.rbenv/versions/2.7.5/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/bin/fastlane:23:in `<top (required)>'\n/Users/vagrant/.rbenv/versions/2.7.5/bin/fastlane:23:in `load'\n/Users/vagrant/.rbenv/versions/2.7.5/bin/fastlane:23:in `<main>'
[!] Could not infer your app's version
Deploy failed, error: exit status 1
```

[Bitrise issue #130](https://github.com/bitrise-steplib/steps-deploy-to-itunesconnect-deliver/issues/130)

**Comment by owurman** [comment]
We're still seeing this in 2.212.2:

`upload_price_tier.rb:24:in `upload': \e[31m[!] undefined method `first' for nil:NilClass\e[0m (NoMethodError) `

**Comment by MortenHN81** [comment]
Still getting this error in 2.212.2

bundler: failed to load command: fastlane (/Users/xx/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/bin/fastlane)
/Users/xx/.rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/fastlane-2.212.2/deliver/lib/deliver/upload_price_tier.rb:24:in `upload': \e[31m[!] undefined method `first' for nil:NilClass (NoMethodError)

      if app_prices.first
                   ^^^^^^\e[0m

**Comment by MiguelSOliveira** [comment]
> Got around it by commenting out price tier from our `deliver` call and setting it manually in ASC. If you are just releasing a new version of your app it should use the existing pricing.

Sorry, just to confirm. You're saying that if the price is set in app store connect, and not in fastlane, it will not overwrite the price and the publish will work?

Because that's good enough for me at the moment.

**Comment by bartpowers** [comment]
> > Got around it by commenting out price tier from our `deliver` call and setting it manually in ASC. If you are just releasing a new version of your app it should use the existing pricing.
> 
> Sorry, just to confirm. You're saying that if the price is set in app store connect, and not in fastlane, it will not overwrite the price and the publish will work?
> 
> Because that's good enough for me at the moment.

Seems to work that way for us.  Give it a shot and check the listing before submitting for review

**Comment by deiv061** [comment]
Still getting this error in 2.212.2
```
 /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/cli_tools_distributor.rb:126:in `ensure in take_off': uninitialized constant FastlaneCore::UpdateChecker (NameError)
from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/fastlane/lib/fastlane/cli_tools_distributor.rb:126:in `take_off'
from /usr/local/lib/ruby/gems/2.7.0/gems/fastlane-2.212.2/bin/fastlane:23:in `<top from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:25:in `load'
from /usr/local/lib/ruby/gems/2.7.0/bin/fastlane:25:in `<main>'
 /usr/local/lib/ruby/site_ruby/2.7.0/rubygems/core_ext/kernel_require.
 ```

**Comment by DETHIEDIOUF4** [comment]
i have just update my  fastlane version  and it's done

**Comment by sonic555gr** [comment]
Hello! 

Based on the comments above, I can't get a clear answer as to if this issue has been fixed on a recent fastlane release or not. We don't have explicit pricing set on our products as they are free apps and nowhere in our fastfile is "prices" mentioned. Yet the error remains. I updated our project to version 2.112.2 and we will be pushing a new version out next week after regression. Based on the comments, I don't feel very confident that 2.112.2 will fix the issue. Can someone give me a more solid answer please?

**Comment by owurman** [comment]
> Hello!
> 
> Based on the comments above, I can't get a clear answer as to if this issue has been fixed on a recent fastlane release or not. We don't have explicit pricing set on our products as they are free apps and nowhere in our fastfile is "prices" mentioned. Yet the error remains. I updated our project to version 2.112.2 and we will be pushing a new version out next week after regression. Based on the comments, I don't feel very confident that 2.112.2 will fix the issue. Can someone give me a more solid answer please?

IIRC it did not work for me with 2.212.2. However, removing the app prices from my flow did fix it. I suggest you look again, as you probably are setting the price somewhere in your workflow.

**Comment by sonic555gr** [comment]
@owurman 

Thank you for your response. There is no mention of prices in our project including fastfile and related scripts. I'm concerned this issue will require a fastlane release. As I mentioned earlier, we are planning an attempted release with the update and I will reply on this thread with my findings. I really home I don't have to result in a forked version for a fix as mentioned above. 

Best

**Comment by PaintThis** [comment]
FYI: we experienced this same problem and updated to fastlane 2.212.2 and that fixed the issue without any other changes.

**Comment by sonic555gr** [comment]
Updating to the latest version has fixed the issue for us

**Comment by Azeem00786** [comment]
@sonic555gr
Thanks for your response,
but i faced this issue only with one client's app rest of my app uploaded successfully.
there are same code and same fastlane and xcode version.
fastlane version- 2.211.0
xcode version-14.2

**Comment by sonic555gr** [comment]
Latest working fastlane version is 2.212.0 please try it that one.

**Comment by Azeem00786** [comment]
i did update fastlane version 2.212.2 instead of 2.211.0 bit the the issue is still persist.
![image](https://github.com/fastlane/fastlane/assets/91094418/c1cdf0f0-32c4-45b8-971a-ed0f43ec516d)

**Comment by gtokman** [comment]
<img width="1018" alt="CleanShot 2023-05-21 at 09 32 39@2x" src="https://github.com/fastlane/fastlane/assets/12258850/ada8c140-ca23-4d07-8a88-bbb72ae33158">

I'm experiencing this issue on the lastest version of fastlane. 

I tried adding `Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')` to the fastfile but didn't have any luck.

**Comment by gholias** [comment]
I'm also having the same problem. Is there a fix yet? I tried adding the `Spaceship::ConnectAPI::App.const_set('ESSENTIAL_INCLUDES', 'appStoreVersions')` but still not able to set the price

**Comment by jijojohn** [comment]
it's working after updating fastlane to 2.213.0

**Comment by christruong90** [comment]
I am still seeing this issue on the new fastlane update 2.213.0:
![image](https://github.com/fastlane/fastlane/assets/19395006/1a5fe1ac-a281-40e7-8baa-f291ed9bb098)

**Comment by memoriasIT** [comment]
For me using version 2.213.0 fixed the issue.
Make sure you are using that version if you have different ones installed.
I used the following line in the Gemfile:
`gem "fastlane", "2.213.0"`

**Comment by hassan-ioki** [comment]
This is till causing issues with version 2.214.0. As our app was free, we just ignored the price updates but please look into a proper fix for this. 🙏

**Comment by Azeem00786** [comment]
i am using fastlane version 2.212.2 and my app was free app but we are are facing the issue. please suggest any proper solution

**Comment by LaGregance** [comment]
I have the same problem since I sent the app for review (not sure if it's related, but it's the 1st time I've sent a new build since)

**Comment by assembleMHN** [comment]
im also having this issue for 13 out of 62 apps released with fastlane, they use the exact same fastlane file, app file and deliver file, its the same app project released to 62 different accounts so its very weird that it would work for some of them and not others.

**Comment by pekq** [comment]
We are also seeing this error and forcing us not to use Fastlane for app updates. Using version `2.214.0` and attempting to update a free app.

**Comment by gavrichards** [comment]
I just hit this. If you comment out `price_tier: 0` as suggested above, you get past this point, but then at the very end of submission you get
> appStoreVersions with id 'XXX' is not in valid state. - This resource cannot be reviewed, please check associated errors to see why.
> App is missing required pricing. - App is not eligible for submission until pricing has been set.

I then manually went into App Store Connect, selected free, all countries, and then everything went through ok.

**Comment by Arpanexe** [comment]
This issue occurred when we change the Base country or Region in pricing and availability section and select he base price for the app. For example by default base country or region is United States and if we change it to United Kingdom and choose the price as £0.00 or any other price, this issue occurs.

**Comment by lakshydeep-14** [comment]
> I just hit this. If you comment out `price_tier: 0` as suggested above, you get past this point, but then at the very end of submission you get
> 
> > appStoreVersions with id 'XXX' is not in valid state. - This resource cannot be reviewed, please check associated errors to see why.
> > App is missing required pricing. - App is not eligible for submission until pricing has been set.
> 
> I then manually went into App Store Connect, selected free, all countries, and then everything went through ok.

then what's the use of using fastlane for automation.
Isn't there any other solution for this.

**Comment by memoriasIT** [comment]
@assembleMHN  It used to be fixed just with the fastlane version, but now I got it again even with 2.214.0 :(

**Comment by memoriasIT** [comment]
@assembleMHN  Make sure you are using a newer version of fastlane
https://github.com/fastlane/fastlane/issues/21125#issuecomment-1619162029

Check your output to make sure you are using that version as well!

**Comment by owurman** [comment]
@assembleMHN We may have actually just had this recur today after having no issues (I think we removed price_tier from our Deliverfile, though). Maybe there's another change with the ASC API? Or a coincidence that we're both seeing it today.

**Comment by owurman** [comment]
Unfortunately, for me at least, it's still an issue (although we haven't updated Fastlane in a few versions). It's been low enough priority that I haven't dug into it, since we just create the ASC entries manually.

**Comment by owurman** [comment]
Actually, I take it back. Our latest issue is https://github.com/fastlane/fastlane/issues/21820

We will update fastlane

**Comment by assembleMHN** [comment]
> > @assembleMHN We may have actually just had this recur today after having no issues (I think we removed price_tier from our Deliverfile, though). Maybe there's another change with the ASC API? Or a coincidence that we're both seeing it today.
> 
> Were you able to get rid from this error? We previously removed the price_tier as well but this error went back recently.

@isamuhiro no sorry, i have not been able to solve it the issue still persist. Only solution for me is to go in to ASC manually and set the price tier, as i work for company doing white label apps, its very annoying as i create lots of apps on app store :(

**Comment by assembleMHN** [comment]
We still have this issue on new apps, first upload of a new app where price is not set, fastlane always breaks with:

"App is missing required pricing. - App is not eligible for submission until pricing has been set.\e[0m"

Its been like this for more than a year now :(
